tests/testthat/test-FilterState.R

```diff
@@ -1,196 +1,180 @@
 # initialize ----
-testthat::test_that("The constructor accepts character as varname", {
-  testthat::expect_no_error(InteractiveFilterState$new(c(7), varname = "test"))
-  testthat::expect_error(InteractiveFilterState$new(c(7), varname = quote(pi)))
-  testthat::expect_error(InteractiveFilterState$new(c(7), varname = call("test")))
+testthat::test_that("constructor checks arguments", {
```

Comment:
I'm thinking these types of tests may not be needed. I feel like they are

1. testing `checkmate` which is already tested
2. potentially brittle

By 'brittle' I mean: let's say we allow one of the arguments to be `NA` in the future that we don't allow now. The associated test would then fail. But I think that's testing details of implementation vs. intended behavior.

In other words, I feel tests like these aren't really testing the behavior of the classes which is what I think tests should focus on.

Curious what others think.

Reply:
Agree with (1), disagree with (2). I think testing assertions is redundant, but I was explicitly asked to do so in another project, so I thought I would go ahead, especially since input types were tested previously. And if we do test for allowed input, I _would_ want a test to fail if another type of input is allowed. This only means the tests did not permit a forbidden input in the first place.